### ISSUE TITLE
Suggestion/request for a crucial NVDA One-Handed Mode Feature regarding Braille Input

### ISSUE THREAD

**Issue by Daredevilsprmcy** [nvaccess/nvda#6960]
I have a feature idea that would greatly benefit people such as myself.
Basically, whenever I buy a braille display, most of the time, it lacks a one-handed mode.
That makes a really expensive equipment useless because the only thing I could do, at that point, is read braille but not type braille.
Now the companies make the same excuse. That it is the screen-reader’s fault.
This means that if a screen-reader doesn’t have one-handed mode for typing braille, the braille display that didn’t come with one-handed mode on board, will not gain it with it.
But this is where I turn to you guys for a big request.
I know that you are currently in the process of creating a way for NVDA users to input contracted braille. I love the idea because I still use the USA Contracted braille code, and the contracted UEB is something I’m still learning.
So, it would be incredible if NV Access came up with a solution for one-handed mode in terms of braille input.
I would have posted this idea on NVDA discussion lists, but I’m afraid of not getting anywhere with it or some one would write a very clunky addon.
Instead, I would prefer to have an official NVDA feature integrated into the screen-reader.
If it does need to be an addon, I still would like it to come officially from you guys.
It could either be free or paid because I think it’s worth paying for.
If NVDA has the ability to not only do braille input in contracted braille but also have a one-handed mode, then it would make any and every braille display be fully accessible for people like me who can only use one hand to type braille.
I’m super excited to hear your view on all this.
If you’re not aware, generally, one-handed mode involves accumulating all the necessary dots for the character 1 key or few at a time, then by hitting a spacebar, to confirm the whole braille character.
For instance: to enter the letter “q” or dots 1,2,3,4,5, in one-handed mode, a person could individually press 1 then 2 then 3 then 4 then 5 followed by a spacebar. He/she may also press dots 1,2,3 together, then dots 4,5 together, then confirm it with a spacebar. To enter a space, one would need to press the spacebar twice. To enter backspace, either press dot 7 followed by space or together dots 7 with spacebar. The same goes for the return key dot 8.
 
Another method is to lock a braille display’s left-hand or right-hand keys for braille entry.
In this method, if it is locked in left-hand one-handed mode, the letter “q” would be entered by pressing dots 1,2,3 together, then dots 1,2 together to comprise the right-hand dots of the braille cell.
A button could be assigned to represent dot 0 if one of the braille cell columns is not required such as letter “l” or dots 1,2,3 followed by a dot 0 since there is no need for the right braille column.
Hopefully something will come out of this.
Keep up the awesome work.

**Comment by jcsteh** [comment]
Thanks for filing this. Do you have thoughts on which method of entry is preferred? The first method (pressing space after all dots are entered) seems easier to understand and I guess also works for users who may not be able to use three fingers simultaneously. On the other hand, the second method (entering the left hand dots then the right hand dots but all on the left side of the keyboard) seems faster, since you don't have to space after every cell. It also requires less hand movement because the hand can just stay on the left side. It does seem difficult to learn and hard to explain, though. One thing we could do is allow the right hand dots to be typed on either the left or the right, but this wouldn't work for users who can't use three fingers simultaneously (if this is something we need to be concerned about; I don't know). Do you have a feeling for which method is more common/preferred? I'd prefer not to have multiple methods, as it makes things confusing, so we should settle on one.

**Comment by jcsteh** [comment]
I guess the first method might be easier to learn, but it's probably very inefficient because of the amount of hand movement it requires. If we're talking about one handed usage, I'm guessing eliminating unnecessary hand movement is a good thing. Is that a valid assumption?

**Comment by Daredevilsprmcy** [comment]
Hello James,

 

To tell you the truth, most of braille displays and notetakers that had one-handed mode available had the first method.

It was easier to learn, but required additional keystrokes to confirm everything with a spacebar.

However, the main difficulty with the second method is that the instructions on how to use it were more complicated than the method itself.

But it does make typing a lot faster.

So to answer your question, the first method with a spacebar confirmation is just more known.

 

Dmitriy Lazarev

**Comment by Daredevilsprmcy** [comment]
Yes, and also eliminating unnecessary hand movements would make NVDA faster in interpreting the keystrokes.

But then again, not all braille displays have an extra key to be a dot 0 key for an empty column.

For example for letter “a”, you need to put dot 1, followed by empty column button.

So even though this method doesn’t need to be confirmed with a spacebar, that empty column button still needs to be pressed.

 

So to make a final decision, let’s just stick to the first method.

Besides, as you’ve pointed out, a person might have difficulty pressing more than one key at a time.

So basically, letter “a” would be dot 1 followed by a spacebar. Letter “w” would be either dot 2, then dots 456, then spacebar or dots 2, then 4, then 5, then 6, then spacebar.

This method gives the user a greater range of motion.

They could even type the separate dots in any combination and then confirm it with a spacebar.

 

Dmitriy Lazarev

**Comment by derekriemer** [comment]
@jcsteh should this be labeled with a priority number? I'm just curious.

**Comment by Adriani90** [comment]
I am closing this in favor of #10674 which contains references to more concrete technical ideas discussed in the braille extender addon repository.
